packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1589,70 +1588,70 @@ class TransformLayer extends OffsetLayer {
   ///
   /// The [transform] property must be non-null before the compositing phase of
   /// the pipeline.
-  Matrix4 get transform => _transform;
-  Matrix4 _transform;
-  set transform(Matrix4 value) {
+  Matrix4? get transform => _transform;
+  Matrix4? _transform;
+  set transform(Matrix4? value) {
     assert(value != null);
-    assert(value.storage.every((double component) => component.isFinite));
+    assert(value!.storage.every((double component) => component.isFinite));
     if (value == _transform)
       return;
     _transform = value;
     _inverseDirty = true;
     markNeedsAddToScene();
   }
 
-  Matrix4 _lastEffectiveTransform;
-  Matrix4 _invertedTransform;
+  Matrix4? _lastEffectiveTransform;
+  Matrix4? _invertedTransform;
   bool _inverseDirty = true;
 
   @override
   void addToScene(ui.SceneBuilder builder, [ Offset layerOffset = Offset.zero ]) {
     assert(transform != null);
-    _lastEffectiveTransform = transform;
+    _lastEffectiveTransform = transform!;
```

Comment:
this `!` is redundant as far as i can tell

Reply:
right. BTW I made an `unnecessary_null_check` lint that should spot this one once available.

Reply:
nice